lib/sensors/mux.cpp

```diff
@@ -0,0 +1,84 @@
+#include "mux.hpp"
+
+namespace hyped::sensors {
+template<typename T, std::size_t N>
+Mux<T, N>::Mux(io::I2c &i2c,
+               const std::uint8_t mux_address,
+               const std::array<std::unique_ptr<II2cMuxSensor<T>>, N> sensors,
+               core::ILogger &log)
+    : log_(log),
+      i2c_(i2c),
+      sensors_(std::move(sensors))
+{
+}
+
+template<typename T, std::size_t N>
+Mux<T, N>::~Mux()
+{
```

Comment:
in light of the other comment, you should static assert here that `N < 8`

Reply:
in the deconstructor and not the constructor?

Reply:
got it, pushing in a sec